pandas/tests/io/formats/style/test_to_latex.py

```diff
@@ -484,8 +484,160 @@ def test_parse_latex_css_conversion(css, expected):
     assert result == expected
 
 
+@pytest.mark.parametrize("environment", ["tabular", "longtable"])
+def test_parse_latex_css_convert_minimal(styler, environment):
+    # parameters ensure longtable template is also tested
+    styler.highlight_max(props="font-weight:bold;")
+    result = styler.to_latex(convert_css=True, environment=environment)
+    assert (
+        dedent(
+            f"""\
+        0 & 0 & \\bfseries -0.61 & ab \\\\
+        1 & \\bfseries 1 & -1.22 & \\bfseries cd \\\\
+        \\end{{{environment}}}
+    """
+        )
+        in result
+    )
+    result = styler.to_latex(convert_css=False, environment=environment)
+    assert (
+        dedent(
+            f"""\
+        0 & 0 & \\font-weightbold -0.61 & ab \\\\
+        1 & \\font-weightbold 1 & -1.22 & \\font-weightbold cd \\\\
+        \\end{{{environment}}}
+    """
+        )
+        in result
+    )
+
+
 def test_parse_latex_css_conversion_option():
     css = [("command", "option--latex--wrap")]
     expected = [("command", "option--wrap")]
     result = _parse_latex_css_conversion(css)
     assert result == expected
+
+
+def test_longtable_comprehensive(styler):
+    result = styler.to_latex(
+        environment="longtable", hrules=True, label="fig:A", caption=("full", "short")
+    )
+    assert result == dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        \\caption[short]{full} \\label{fig:A} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endfirsthead
+        \\caption[]{full} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endhead
+        \\midrule
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\midrule
+        \\endfoot
+        \\bottomrule
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+
+
+def test_longtable_minimal(styler):
+    result = styler.to_latex(environment="longtable")
+    assert result == dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        {} & {A} & {B} & {C} \\\\
+        \\endfirsthead
+        {} & {A} & {B} & {C} \\\\
+        \\endhead
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\endfoot
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+
+
+def test_longtable_multiindex_columns(df):
```

Comment:
I suggest that this test function is split into two functions.
It will be necessary to duplicate two lines (572-573), but it will be more readable and maintainable, IMHO.

Reply:
parametrized this instead.